src/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -759,5 +759,52 @@ public void CanBeFinalized()
             GC.WaitForPendingFinalizers();
             Assert.True(FinalizingPing.WasFinalized);
         }
-    }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task SendPingAsyncWithHostAndTtlAndFragmentPingOptions(bool fragment)
+        {
+            IPAddress localIpAddress = await TestSettings.GetLocalIPAddressAsync();
+
+            byte[] buffer = TestSettings.PayloadAsBytes;
+
+            PingOptions  options = new PingOptions();
+            options.Ttl = 32;
+            options.DontFragment = fragment;
+
+            await SendBatchPingAsync(
+                (ping) => ping.SendPingAsync(TestSettings.LocalHost, TestSettings.PingTimeout, buffer, options),
+                (pingReply) =>
+                {
+                    PingResultValidator(pingReply, localIpAddress);
+                });
+        }
+
+        [Fact]
+        [OuterLoop] // Depends on external host and assumption that network respects and does not change TTL
+        public async Task SendPingToExternalHostWithLowTtlTest()
+        {
+            string host = System.Net.Test.Common.Configuration.Ping.PingHost;
+            PingOptions options = new PingOptions();
+            bool reachable = false;
+
+            Ping ping = new Ping();
+            for (int i = 0; i < s_pingcount; i++)
+            {
+                PingReply pingReply = await ping.SendPingAsync(host, TestSettings.PingTimeout, TestSettings.PayloadAsBytesShort);
+                if (pingReply.Status == IPStatus.Success)
+                {
+                    reachable = true;
+                    break;
+                }
+            }
+            if (reachable) {
```

Comment:
It looks like if all the pings fail, we won't validate anything?  Is that by design?

Reply:
yes, that was intentional. I did not want test to fail if firewall block ping or if there is any reachability problem.
I modified the test and it will now properly report is test was not executed:
```
export COREFX_NET_PING_HOST=corefx-net.cloudapp.net
../../../../eng/common/msbuild.sh /t:rebuildandtest /p:outerloop=true

...

      System.Net.NetworkInformation.Tests.PingTest.SendPingToExternalHostWithLowTtlTest [STARTING]
      System.Net.NetworkInformation.Tests.PingTest.SendPingToExternalHostWithLowTtlTest [SKIP]
        Host corefx-net.cloudapp.net is not reachable. Skipping test.
      System.Net.NetworkInformation.Tests.PingTest.SendPingToExternalHostWithLowTtlTest [FINISHED] Time: 40.0844645s
```